### PR TITLE
[ZEPPELIN-5167] Set HOME env variable to in Dockerfile, so Helium can install package successfully

### PR DESCRIPTION
**Pull request by Leemoonsoo**
### What is this PR for?
Without HOME env variable set in Docker image, installing a helium visualization package fails with error.

```
 INFO [2021-03-04 06:40:51,674] ({qtp418304857-31} YarnTaskExecutor.java[execute]:54) - Running 'yarn config set cache-folder /opt/zeppelin/local-repo/helium-bundle/yarn-cache --registry=https://registry.npmjs.org/' in /opt/zeppelin/local-repo/helium-bundle
ERROR [2021-03-04 06:40:51,793] ({Thread-24} InputStreamHandler.java[log]:60) - /opt/zeppelin/local-repo/helium-bundle/node/yarn/dist/node_modules/mkdirp/index.js:90
ERROR [2021-03-04 06:40:51,795] ({Thread-24} InputStreamHandler.java[log]:60) -                     throw err0;
ERROR [2021-03-04 06:40:51,795] ({Thread-24} InputStreamHandler.java[log]:60) -                     ^
ERROR [2021-03-04 06:40:51,795] ({Thread-24} InputStreamHandler.java[log]:60) - 
ERROR [2021-03-04 06:40:51,796] ({Thread-24} InputStreamHandler.java[log]:60) - Error: EACCES: permission denied, mkdir '/.cache'
ERROR [2021-03-04 06:40:51,796] ({Thread-24} InputStreamHandler.java[log]:60) -     at Error (native)
ERROR [2021-03-04 06:40:51,796] ({Thread-24} InputStreamHandler.java[log]:60) -     at Object.fs.mkdirSync (fs.js:922:18)
ERROR [2021-03-04 06:40:51,797] ({Thread-24} InputStreamHandler.java[log]:60) -     at sync (/opt/zeppelin/local-repo/helium-bundle/node/yarn/dist/node_modules/mkdirp/index.js:71:13)
ERROR [2021-03-04 06:40:51,797] ({Thread-24} InputStreamHandler.java[log]:60) -     at Function.sync (/opt/zeppelin/local-repo/helium-bundle/node/yarn/dist/node_modules/mkdirp/index.js:77:24)
ERROR [2021-03-04 06:40:51,797] ({Thread-24} InputStreamHandler.java[log]:60) -     at Object.<anonymous> (/opt/zeppelin/local-repo/helium-bundle/node/yarn/dist/bin/yarn.js:30:8)
ERROR [2021-03-04 06:40:51,797] ({Thread-24} InputStreamHandler.java[log]:60) -     at Module._compile (module.js:570:32)
ERROR [2021-03-04 06:40:51,798] ({Thread-24} InputStreamHandler.java[log]:60) -     at Object.Module._extensions..js (module.js:579:10)
ERROR [2021-03-04 06:40:51,799] ({Thread-24} InputStreamHandler.java[log]:60) -     at Module.load (module.js:487:32)
ERROR [2021-03-04 06:40:51,801] ({Thread-24} InputStreamHandler.java[log]:60) -     at tryModuleLoad (module.js:446:12)
ERROR [2021-03-04 06:40:51,801] ({Thread-24} InputStreamHandler.java[log]:60) -     at Function.Module._load (module.js:438:3)
```

This error can be solved by manually adding HOME env variable when starting docker container, like

```
docker run -p8080:8080 --rm -e HOME=/opt/zeppelin -e ZEPPELIN_HELIUM_REGISTRY=helium,https://s3.amazonaws.com/helium-package/helium.json apache/zeppelin:0.9.0
```

However, the error message is not very intuitive and it may take lots of time to find a solution.
So, I think it make sense to set the HOME env variable in Dockerfile.

### What type of PR is it?
Bug Fix

### What is the Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-5167

### How should this be tested?

Reproduce the problem by running Zeppelin from the latest docker image without the HOME env variable.
And try to install a visualization package from Helium menu

```
docker run -p8080:8080 --rm -e ZEPPELIN_HELIUM_REGISTRY=helium,https://s3.amazonaws.com/helium-package/helium.json apache/zeppelin:0.9.0
```

And then test with HOME environment variable, and verify visualization package is installed without error.
```
docker run -p8080:8080 --rm -e HOME=/opt/zeppelin -e ZEPPELIN_HELIUM_REGISTRY=helium,https://s3.amazonaws.com/helium-package/helium.json apache/zeppelin:0.9.0
```

### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? no

### CODE REVIEW

**Comment by Reamer** [comment]
I think this is the first step in the right direction. I think we should replace `Z_HOME` completely with `HOME`.

**Comment by Leemoonsoo** [comment]
> I think this is the first step in the right direction. I think we should replace Z_HOME completely with HOME.

How about rename `Z_HOME` to `ZEPPELIN_HOME`, and keep the `HOME` env as well?
The reason is, `ZEPPELIN_HOME` is already being used in many scripts like [bin/zeppelin.sh](https://github.com/apache/zeppelin/blob/master/bin/zeppelin.sh).

And `HOME` is still required to make some external command work (`yarn` cmd for installing Helium package),
but doesn't necessarily have to be the same as `ZEPPELIN_HOME`.

What do you think?

**Comment by Reamer** [comment]
Sounds good to me

**Comment by Leemoonsoo** [comment]
@Reamer Replaced Z_HOME with ZEPPELIN_HOME. Can you help review this?

File: scripts/docker/zeppelin-server/Dockerfile [hunk 1]
```diff
@@ -41,43 +41,43 @@ ENV LANG=en_US.UTF-8 \
     LC_ALL=en_US.UTF-8 \
     JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64 \
     VERSION="${version}" \
-    Z_HOME="/opt/zeppelin" \
+    ZEPPELIN_HOME="/opt/zeppelin" \
     ZEPPELIN_ADDR="0.0.0.0" \
```
Comment (Reamer) [inline scripts/docker/zeppelin-server/Dockerfile:45]: @Leemoonsoo 
I think you should also set `HOME` in this place to fix your problem with Helium in this Dockerfile as well.

**Comment by Reamer** [comment]
> @Reamer Replaced Z_HOME with ZEPPELIN_HOME. Can you help review this?

The replacement looks good. I have only added a small comment.

File: scripts/docker/zeppelin-server/Dockerfile (hunk 1 above)
Comment (Leemoonsoo) [reply scripts/docker/zeppelin-server/Dockerfile:46]: right. I updated this, too.

**Comment by Leemoonsoo** [comment]
Thank @Reamer for review this!

I'll merge if there's no further comment.

**Pull request closed without merging**